core/new-gui/src/app/workspace/component/result-panel/console-frame/console-frame.component.ts

```diff
@@ -13,11 +14,26 @@ import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
 })
 export class ConsoleFrameComponent implements OnInit, OnChanges {
   @Input() operatorId?: string;
+  @ViewChild(CdkVirtualScrollViewport) viewPort?: CdkVirtualScrollViewport;
+  workerAbbrToLongNameMapping = new Map();
+
   // display error message:
   errorMessages?: Readonly<Record<string, string>>;
 
   // display print
-  consoleMessages: ReadonlyArray<string> = [];
+  consoleMessages: ReadonlyArray<PythonConsoleMessage> = [];
+
+  showTimestamp: boolean = true;
+
+  labelMapping = new Map([["PRINT", "default"]]);
+
+  // TODO: find a better way to generate different color for each worker.
+  workerColorMapping = new Map([
+    ["Worker-0", "lime"],
+    ["Worker-1", "cyan"],
+    ["Worker-2", "volcano"],
+    ["Worker-3", " orange"],
```

Comment:
try something like this:

```
import { presetPalettes } from '@ant-design/colors';


  getWorkerColor(workerIndex: number): string {
    const presetPalettesSize = Object.keys(presetPalettes).length
    const colorKey = Object.keys(presetPalettes)[workerIndex % presetPalettesSize]
    const color = presetPalettes[colorKey][Math.floor(workerIndex / presetPalettesSize)]
    return color
  }
  
  getWorkerIndex(workerId: string): number {
    const tokens = workerId.split("-");
    return parseInt(tokens.at(tokens.length - 1) || "0");
  }
```

the current `presetPalettes` has 12 color keys, each color key has 10 shades of colors, so we can have up to 120 workers without repeated color.  See https://ant.design/docs/spec/colors#neutral-color-palette

Also we can use only 10 color keys, if you want the colors to be similar every 10 workers, (like workers 0, 10, 20, ... have similar colors)

Reply:
took the suggestion with `presetPalettes`. thanks.